rules/bad_defer.go

```diff
@@ -0,0 +1,58 @@
+package rules
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/securego/gosec"
+)
+
+type badDefer struct {
+	gosec.MetaData
+	typ     string
```

Comment:
I would make this a bit more generic to support multiple types and methods.

```Go
type deferType struct {
    type string
    methods []string
}
```
and use it as:

```Go
type badDefer struct {
  gosec.MetaData
  types []deferType
}
```

Reply:
Changes made.